eessi/reframe/eessi_utils/hooks.py

```diff
@@ -93,3 +93,46 @@ def assign_one_task_per_gpu(test: rfm.RegressionTest) -> rfm.RegressionTest:
     test.num_gpus_per_node = num_gpus_per_node
     test.num_tasks_per_node = num_tasks_per_node
     test.num_tasks = test.num_nodes * num_tasks_per_node
+
+
+def filter_tests_by_device_type(test: rfm.RegressionTest, device_type: str):
+    """
+    filter valid_systems by device type,
+        unless specified with --setvar valid_systems=<comma-separated-list>
+    """
+    if not test.valid_systems:
+        is_cuda_module = utils.is_cuda_required_module(test.module_name)
+        valid_systems = ''
+
+        if is_cuda_module and device_type == 'gpu':
+            # CUDA modules and when using a GPU for non-bonded interactions require partitions with 'gpu' feature
+            valid_systems = '+gpu'
```

Comment:
Should use `'+%s' % eessi_utils.features.GPU_GENERIC`

Reply:
that's also for a separate PR